### ISSUE TITLE
Map frame coordinates not exported to PDF if they have a negative spacing and a map layer render mode is set to multiply

### ISSUE THREAD

**Issue by mikedufty** [qgis/QGIS#52382]
### What is the bug or the crash?

If draw coordinates is enabled for a map grid in a print layout, but with a negative distance to map frame this displays the labels outside the frame in the layout composer, but when the layout is exported to image, the coordinates do not appear in the exported image.
The issue only occurs if a layer in the map item is set to a render mode of multiply.

It may seem unusual to wish to show 'inside' labels outside the frame, but it was a very useful workaround for issue  #47250 Print layout: crop to content produces incorrect cropping when grid coordinates labels are present.

### Steps to reproduce the issue

1. Create a new map and a new print layout
2. Add a layer to the map eg. built in world layer and set render mode to multiply.
3. Add item - map into layout set extent to a reasonable view eg Australai
4. Add grid to map
5. Modify grid
6. tick enable grid, set interval so there is at least one grid line (eg 10 degrees for Australia map).
7. tick draw coordinates
8. set left coordinates to 'inside frame'
9. Set distance to map frame in draw coordinates dialogue to -10mm

![image](https://user-images.githubusercontent.com/34697905/227698630-560089ab-6ea7-4c8f-b5cd-4c27f1eb6d76.png)
10. Go to layout export as image, select jpg and a filename and location.  Dialogue can be left at defaults.

![image](https://user-images.githubusercontent.com/34697905/227703116-3019cae3-3d09-4db7-9e94-325c7107dee3.png)

11. Open resulting file and you will see the portions of the coordinate outside the frame have been cropped.
![test](https://user-images.githubusercontent.com/34697905/227703143-39f396dc-0447-4d64-9e82-8658b1dc9285.jpg)
12.  Change blending mode to normal and repeat and the coordinate will not be cropped.

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.30.0-'s-Hertogenbosch | QGIS code revision | f186b8efe0
-- | -- | -- | --
Qt version | 5.15.3
Python version | 3.9.5
GDAL/OGR version | 3.6.3
PROJ version | 9.2.0
EPSG Registry database version | v10.082 (2023-02-06)
GEOS version | 3.11.2-CAPI-1.17.2
SQLite version | 3.41.1
PDAL version | 2.5.2
PostgreSQL client version | unknown
SpatiaLite version | 5.0.1
QWT version | 6.1.6
QScintilla2 version | 2.13.1
OS version | Windows 10 Version 2009
  |   |   |  
Active Python plugins
autoSaver | 2.6
changeDataSource | 3.1
FreehandRasterGeoreferencer | 0.8.3
latlontools | 3.6.9
MemoryLayerSaver | 4.0.4
processing_saga_nextgen | 0.0.7
qfieldsync | v4.4.8
QuickOSM | 2.1.1
valuetool | 3.0.15
ViewshedAnalysis | 1.8
volume_calculation_tool | 0.4
db_manager | 0.1.20
grassprovider | 2.12.99
MetaSearch | 0.3.6
processing | 2.12.99

</body></html>QGIS version
3.30.0-'s-Hertogenbosch
QGIS code revision
[f186b8efe0](https://github.com/qgis/QGIS/commit/f186b8efe0)
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.6.3
PROJ version
9.2.0
EPSG Registry database version
v10.082 (2023-02-06)
GEOS version
3.11.2-CAPI-1.17.2
SQLite version
3.41.1
PDAL version
2.5.2
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
autoSaver
2.6
changeDataSource
3.1
FreehandRasterGeoreferencer
0.8.3
latlontools
3.6.9
MemoryLayerSaver
4.0.4
processing_saga_nextgen
0.0.7
qfieldsync
v4.4.8
QuickOSM
2.1.1
valuetool
3.0.15
ViewshedAnalysis
1.8
volume_calculation_tool
0.4
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by agiudiceandrea** [comment]
@mikeduffy, it seems to me the reported incorrect behaviour does occur setting any blending mode, for layer or for feature, different from "Normal". Could you check?

**Comment by mikeduffy** [comment]
I don't think I'm the Mike Duffy you're looking for.  Sorry about that.  I
have no involvement with anything doing what you described.

Mike
www.mikeduffy.com

**Comment by mikedufty** [comment]
Yes, I will see if I can find time to try that.  Note my user name has a T in it.  MIKEDUFTY hence the puzzled response from Mike Duffy

**Comment by mikedufty** [comment]
I have done some further testing and can confirm that the issue occurs with every blending mode except normal.
I also tried an export to PDF and the issue occurs there as well.

I previously checked having the layer with alternate rendering turned on, but outside the extent of the layout map object, and the issue still occurs in that situation.
